src/System.Net.Security/src/System/Net/Security/TlsCipherSuiteNameParser.ttinclude

```diff
@@ -0,0 +1,584 @@
+    Camellia128Gcm,
+    Camellia256Gcm,
+    ChaCha20,
+    ChaCha20Poly1305,
+    Seed,
+    Idea,
+    Aria,
+    Aria128,
+    Aria256,
+    Aria128Gcm,
+    Aria256Gcm,
+}
+
+public enum HashAlgorithmType
+{
+    None,
+    Md5,
+    Sha1,
+    Sha256,
+    Sha384,
+    Sha512,
+    Aead,
+}
+
+internal static class EnumHelpers
+{
+    public static string ToFrameworkName(ExchangeAlgorithmType val)
+    {
+        switch (val)
+        {
+            case ExchangeAlgorithmType.None:
+                return "None";
+            case ExchangeAlgorithmType.Rsa:
+                return "RsaKeyX";
+            case ExchangeAlgorithmType.DiffieHellmanEphermal:
+            case ExchangeAlgorithmType.DiffieHellmanStatic:
+            case ExchangeAlgorithmType.ECDiffieHellman:
+            case ExchangeAlgorithmType.ECDiffieHellmanEphermal:
+                return "DiffieHellman";
+            case ExchangeAlgorithmType.Kerberos5:
+            case ExchangeAlgorithmType.PSK:
+            case ExchangeAlgorithmType.SRP:
+            case ExchangeAlgorithmType.ECCPWD:
+            case ExchangeAlgorithmType.Any:
+                return  "None";
+            default:
+                throw new Exception($"Value `{val}` does not have framework corresponding name.");
+        }
+    }
+
+    public static string ToFrameworkName(CipherAlgorithmType val)
+    {
+        switch (val)
+        {
+            case CipherAlgorithmType.Null:
+            case CipherAlgorithmType.Rc2:
+            case CipherAlgorithmType.Rc4:
+            case CipherAlgorithmType.Des:
+            case CipherAlgorithmType.TripleDes:
+            case CipherAlgorithmType.Aes128:
+            case CipherAlgorithmType.Aes256:
+                return val.ToString();
+            case CipherAlgorithmType.Aes128Gcm:
+            case CipherAlgorithmType.Aes128Ccm:
+            case CipherAlgorithmType.Aes128Ccm8:
+                return "Aes128";
+            case CipherAlgorithmType.Aes256Gcm:
+            case CipherAlgorithmType.Aes256Ccm:
+            case CipherAlgorithmType.Aes256Ccm8:
+                return "Aes256";
+            case CipherAlgorithmType.Idea:
+            case CipherAlgorithmType.Seed:
+            case CipherAlgorithmType.Camellia128:
+            case CipherAlgorithmType.Camellia128Gcm:
+            case CipherAlgorithmType.Camellia256:
+            case CipherAlgorithmType.Camellia256Gcm:
+            case CipherAlgorithmType.ChaCha20Poly1305:
+            case CipherAlgorithmType.Aria128:
+            case CipherAlgorithmType.Aria128Gcm:
+            case CipherAlgorithmType.Aria256:
+            case CipherAlgorithmType.Aria256Gcm:
+                return "None";
+            default:
+                throw new Exception($"Value `{val}` does not have framework corresponding name.");
+        }
+    }
+
+    public static string ToFrameworkName(HashAlgorithmType val)
+    {
+        return val == HashAlgorithmType.Aead ? "None" : val.ToString();
```

Comment:
Consider expanding this, and doing the open-drain-throw default.

Reply:
currently we have 1:1 mapping except Aead so will leave it as is for simplicity

Reply:
I'm not suggesting changing it to be anything complicated, just explicit-case on the 1:1 mapping (returning `val.ToString()` still) and letting Aead be None, and anything future throw so whoever is making the change figures out what to do about it.

The reason I brought it up is I think everything else currently throws on new kinds of data, so someone might hit a deceptive "I'm done" state instead of being really done.

Reply:
will add it it in my ciphersuite policy PR